### PR TITLE
[FX-4492] Remove unwanted tag transitions

### PR DESCRIPTION
**Pull request by mkrl**
[FX-4492](https://toptal-core.atlassian.net/browse/FX-4492)

### Description

Remove the default transitions from all `Tag` components.
I was not able to disable them in a single place (in theory, it should be possible to use `overrides` in `PicassoProvider`, however, in practice it just didn't work at all, probably due to class overrides inside components).

### How to test

- [Temploy](https://picasso.toptal.net/fx-4492-tag-components-have-leftover-styles-from-mui)
- You can use the code below to test the transition, both on master and temploy
```
import React, { useState } from 'react'
import { Button, Container, Tag } from '@toptal/picasso'

const Example = () => {
    const [type, setType] = useState('red')

    return (
        <Container flex>
            <Button
                onClick={() =>
                    setType(prevType => (prevType === 'green' ? 'red' : 'green'))
                }
                size='small'
            >
                Change color
            </Button>
            <Tag.Rectangular variant={type}>Tag</Tag.Rectangular>
        </Container>
    )
}

export default Example

```

### Screenshots

**Before**

https://github.com/toptal/picasso/assets/18409292/05409c53-22fc-4384-83b9-2868a5878a8d

**After**

https://github.com/toptal/picasso/assets/18409292/892ccfd2-1722-4ddc-9111-42860644170e

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [n/a] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4492]: https://toptal-core.atlassian.net/browse/FX-4492?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/lovely-pumpkins-help.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@toptal/picasso': patch
+---
+
+### Tag, Tag.Reactangular
```
Comment (sashuk) [inline .changeset/lovely-pumpkins-help.md:5]:
```suggestion
### Tag, Tag.Rectangular
```

File: .changeset/lovely-pumpkins-help.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+- removed unwanted transitions for background-color and box-shadow
```
Comment (sashuk) [inline .changeset/lovely-pumpkins-help.md:7]: NIT

```suggestion
- remove unwanted transitions for background-color and box-shadow
```

**Review by sashuk — APPROVED** [review]
Tested on temploy, looks great!

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

**Review by TomasSlama — APPROVED** [review]

**Pull request merged**